euphonic/force_constants.py

```diff
@@ -589,6 +609,12 @@ def _calculate_phonons_at_qpts(
             cell_vectors = self.crystal._cell_vectors
             recip_vectors = self.crystal.reciprocal_cell().to(
                 '1/bohr').magnitude
+            # Get conj transpose - the dynamical matrix calculated in C
+            # is passed to Fortran libs so uses Fortran ordering, make
+            # sure reciprocal correction matches this. This only makes
+            # a very small difference and can only be seen in near-flat
+            # mode gradients, but should be corrected anyway
+            recip_asr_correction =  recip_asr_correction.conj().T

```

Comment:
Why is a C-vs-Fortran-ordering problem solved by taking the _conjugate_ transpose?

Reply:
If `recip_asr_correction` is Hermitian, than taking the complex conjugate _is_ taking the transpose. (And taking both is an identity operation.)

If it is symmetric (as you say) and _not_ Hermitian then this is reversing the phase; is there somewhere else in the maths that this phase reversal should be reflected? Indexing in the "wrong" order for a phase reversal seems like a Hermitian thing.

Reply:
The reciprocal ASR correction is derived from the dynamical matrix at gamma, the dynamical matrix is Hermitian, but at gamma it is real (in practice numerically _almost_). To get the reciprocal ASR correction the gamma-point eigenvalues multiplied by the gamma-point eigenvectors and their conjugates of the 3 acoustic modes are subtracted from the gamma-point dynamical matrix to artificially shift the eigenvalues to 0. I'm not sure how this affects the symmetry/Hermicity of the resultant reciprocal ASR correction though... examining the correction this appears to make it exactly symmetric, but I'm not competent enough in the Maths to say why

Reply:
I see! So in an ideal world none of these operations would do anything but in practice we are shuffling numerical errors around and they need to be lined up with their corresponding elements.

That makes it a bit harder for me to naively reason about how things should behave under transformation :sweat_smile: 
If you're happy that the implementation is now following the described method then all is good.